### DISCUSSION TITLE
PostgreSql: get numbers for count

### DISCUSSION THREAD

**Discussion by sualko** [mikro-orm/mikro-orm#6068] [Q&A]
The `pg` driver will return strings for count queries, because of [bigint issues](https://github.com/brianc/node-postgres/issues/378). To disable this behavior you can enable [int8 parsing by default](https://github.com/brianc/node-postgres/pull/427), but I found no way to do this in mikro-orm. I ended up extending the default pg driver and adding the parsing there. Any idea how this can be simplified?

```
class Int8PostgreSqlConnection extends PostgreSqlConnection {
  public override getConnectionOptions(): Knex.PgConnectionConfig {
    const config = super.getConnectionOptions();

    (config.types as TypeOverrides).setTypeParser(20, 'text', parseInt);

    return config;
  }
}

export class Int8PostgreSqlDriver extends AbstractSqlDriver<Int8PostgreSqlConnection> {
  public constructor(config: Configuration) {
    super(config, new PostgreSqlPlatform(), Int8PostgreSqlConnection, [
      'knex',
      'pg',
    ]);
  }
}

```

**Comment by B4nan** [comment]
But if you use `em.count()` or `qb.count()` we normalize to numbers I believe? This is a problem only when you work with the raw data?

**Reply by sualko** [reply]
@B4nan at least I encountered no issues in our integrations tests with `count()` so far, therefore I believe it is only related to raw queries. I'm currently not in the office, therefore I can't test it. Do you want me to create a test repo next week?

**Reply by B4nan** [reply]
i dont need a repro for that, its expected behavior if we are talking about raw data - since the counts are bigints, and those are represented by strings on raw level.

**Reply by B4nan** [reply]
> To disable this behavior you can enable https://github.com/brianc/node-postgres/pull/427,

i would be careful with this, it will affect all bigints, and you will potentially lose precision.
